msal/oauth2cli/authcode.py

```diff
@@ -33,9 +33,34 @@ def obtain_auth_code(listen_port, auth_uri=None):  # Historically only used in t
             ).get("code")
 
 
+def is_wsl():
+    # "Official" way of detecting WSL: https://github.com/Microsoft/WSL/issues/423#issuecomment-221627364
+    # Run `uname -a` to get 'release' without python
+    #   - WSL 1: '4.4.0-19041-Microsoft'
+    #   - WSL 2: '4.19.128-microsoft-standard'
+    import platform
+    uname = platform.uname()
+    platform_name = getattr(uname, 'system', None).lower()
+    release = getattr(uname, 'release', None).lower()
```

Comment:
These 2 lines did not feel right. The default value `None` won't have the `lower()` method. Let's use this chance to make it compatible with some [other versions of Python](https://docs.python.org/2.7/library/platform.html#platform.uname) which do not return uname as a namedtuple.

```suggestion
    platform_name = getattr(uname, 'system', uname[0]).lower()
    release = getattr(uname, 'release', uname[2]).lower()
```